x-pack/plugins/enterprise_search/public/applications/shared/telemetry/telemetry_logic.test.ts

```diff
@@ -35,13 +35,17 @@ describe('Telemetry logic', () => {
       });
     });
 
-    it('throws an error if the telemetry endpoint fails', () => {
+    it('throws an error if the telemetry endpoint fails', async () => {
       mockHttpValues.http.put.mockImplementationOnce(() => Promise.reject());
-      try {
-        TelemetryLogic.actions.sendTelemetry({ action: '', metric: '', product: '' });
-      } catch (e) {
-        expect(e.message).toEqual('Unable to send telemetry');
-      }
+
+      // Using TelemetryLogic.actions.sendTelemetry causes this test to not work -
```

Comment:
I mentioned this in a comment in Slack, but repeating it here. I have this weird sense of guilt about this comment, like we're blaming Kea for doing something weird 😭 

Actions are like **events** in Kea, and reducers and listeners are like **event handlers**.

I would not _expect_ that triggering an event would return a value from an event handler. Therefore, I would not _expect_ that dispatching an action would return a value from a listener.

I mean think bout it, there could be multiple event listeners listening for that event, how would you even know which value to return from the action invocation if there are 5 different handlers or listeners running.

The same is true with Kea and Redux.

Kea abstracts some things away from Redux... like when you call an action in Kea, you are invoking it in a way that it _appears_ like a direct invocation of the handler logic, and less like triggering an event:

```
sendTelemetry()
```

In Redux, it looks more like the following, which makes it less tempting to expect a return value:
```
store.dispatch({ type: 'sendTelemetry' })
```

This is basically a long winded way of saying if you shift your thinking a bit, this seems less like interference.

Reply:
Thanks for taking the time to write that all out Jason!!! Totally agreed and I found that super helpful / learned a few new things.